crates/starfish/core/src/transactions_synchronizer.rs

```diff
@@ -133,19 +144,33 @@ impl InflightTransactionsMap {
         // Now mark all the transactions as fetched from the map
         let mut transactions_to_fetch = self.inner.lock();
         for block_ref in block_refs {
-            let authorities = transactions_to_fetch
-                .get_mut(block_ref)
-                .expect("Should have found a non empty map");
-
-            assert!(authorities.remove(&peer), "Peer index should be present!");
-
-            // if the last one then just clean up
-            if authorities.is_empty() {
-                transactions_to_fetch.remove(block_ref);
+            // Due to periodic cleaning of the map, it is possible that the
+            // block_ref is not present in the map anymore, so we just skip it.
+            if let Some(authorities) = transactions_to_fetch.get_mut(block_ref) {
+                authorities.remove(&peer);
+
+                // If the last one then just clean up
+                if authorities.is_empty() {
+                    transactions_to_fetch.remove(block_ref);
+                }
             }
         }
     }
 
+    /// Cleans up the old locks that are older than `max_age`. This is useful
+    /// to remove the locks that are not used anymore. The latter could happen
+    /// when the timeout occurred and the locks are forgotten to be
+    /// unlocked.
+    fn cleanup_old_locks(self: &Arc<Self>, max_age: Duration) {
```

Comment:
How can it happen that it's forgotten? If we remove it from the map here, is `drop()` called at some later point in time? If not then this could indicate that we might have memory leak as the non-dropped object is still held in memory. Can't we drop the TransactionGuard when the request times out instead of cleaning the whole map periodically?